src/components/PhotoZoom.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useMemo, useEffect, useState, useCallback } from "react"
+import useScrollBlock from "../hooks/useScrollBlock"
+import { TransformWrapper, TransformComponent } from "react-zoom-pan-pinch"
+
+import { BaseProps } from "../data/types"
+interface PhotoZoomProps extends BaseProps {
+  photo: string
+  isLoading: boolean | undefined
+  exit: () => void
+  isTransitioningIn: boolean
+  isTransitioningOut: boolean
+}
+
+function PhotoZoom({
+  photo,
+  isLoading,
+  exit,
+  isTransitioningIn,
+  isTransitioningOut,
+}: PhotoZoomProps) {
+  const [blockScroll, allowScroll] = useScrollBlock()
+  const [container, setContainer] = useState<HTMLDivElement | null>(null)
+
+  const [containerWidth, setContainerWidth] = useState<number>(0)
+  const [containerHeight, setContainerHeight] = useState<number>(0)
+
+  const [imageNaturalWidth, setImageNaturalWidth] = useState<number>(0)
+  const [imageNaturalHeight, setImageNaturalHeight] = useState<number>(0)
+  const alt = "example"
```

Comment:
Let's make the photo alt `photo.description || photo.title || photo.filename`. You can either pass the whole photo model or calculate the `alt` in the parent component and pass it as a prop.